explainability-core/src/test/java/org/kie/trustyai/explainability/local/shap/ShapKernelExplainerTest.java

```diff
@@ -853,4 +855,51 @@ void testCategoricalCollisions() throws ExecutionException, InterruptedException
         assertEquals(1, shapResults.getSaliencies()[1].getPerFeatureImportance().get(1).getScore(), 1e-6);
     }
 
+    @Test
+    void testToString() throws ExecutionException, InterruptedException, TimeoutException {
+        Random random = new Random(0L);
+
+        List<PredictionInput> background = new ArrayList<>();
+        for (int bg = 0; bg < 1; bg++) {
+            List<Feature> features = new ArrayList<>();
+            for (int i = 0; i < 5; i++) {
+                if (i == 2) {
+                    features.add(new Feature("Feature " + i, Type.CATEGORICAL, new Value("B")));
+                } else {
+                    features.add(new Feature("Feature " + i, Type.NUMBER, new Value(0)));
+                }
+            }
+            background.add(new PredictionInput(features));
+        }
+        ShapConfig shapConfig = testConfig.copy()
+                .withBackground(background)
+                .build();
+        ShapKernelExplainer ske = new ShapKernelExplainer(shapConfig);
+
+        List<Feature> features = new ArrayList<>();
+        for (int i = 0; i < 5; i++) {
+            if (i == 2) {
+                features.add(new Feature(
+                        "Feature " + i,
+                        Type.CATEGORICAL,
+                        new Value("A"),
+                        false,
+                        CategoricalFeatureDomain.create(List.of("A", "B"))));
+            } else {
+                features.add(new Feature("Feature " + i, Type.NUMBER, new Value(i)));
+            }
+        }
+        PredictionInput input = new PredictionInput(features);
+        PredictionProvider model = TestUtils.getTwoOutputSemiCategoricalModel(2);
+        PredictionOutput output = model.predictAsync(List.of(input))
+                .get(Config.INSTANCE.getAsyncTimeout(), Config.INSTANCE.getAsyncTimeUnit())
+                .get(0);
+        Prediction prediction = new SimplePrediction(input, output);
+        ShapResults results = ske.explainAsync(prediction, model)
+                .get(Config.INSTANCE.getAsyncTimeout(), Config.INSTANCE.getAsyncTimeUnit());
+        String table = results.toString();
+        //System.out.println(table);
+        assertTrue(true);
```

Comment:
As in, these functions are implicitly what define the correct result

Reply:
I do see the point, however one such a check allows for capturing changes in the generated outputs that may happen in the future (e.g., because of dropped / added inner objects that are part of the `toString`) and also allows us to more easily assess how much we "like" the generated `String`.
Anyway I'm also fine just checking the result is not `null` at the very least.

Reply:
Ok I missed this comment so I made a similar one :D